Add kyt-dlm-devinfo-static

KlausPopp opened the pull request:
Add kyt-dlm-devinfo-static: A process that runs on the device.
Acts as a Iot device client to write static device information in the device twin.
Currently reads firmware version information from /etc/issue.
Device connection string must be configured in a config file /data/kyt/dlm.yaml:
E.g. `device_connection_string: HostName=ci4rail-eval-iot-hub.azure-devices.net;DeviceId=verdinDevBoardOnGecko;SharedAccessKey=2HUd...A=`

KlausPopp committed:
first test of kyt-dlm-devinfo-static

KlausPopp committed:
Added kyt-dlm-devinfo-static MakefileDockerfile dobi

KlausPopp committed:
kyt-dlm-devinfo-static: add debugs

KlausPopp committed:
kyt-dlm-devinfo-static: add debugs

KlausPopp committed:
kyt-dlm-devinfo-static: iotedge=false

KlausPopp committed:
kyt-dlm-devinfo-static: add debugs

KlausPopp committed:
kyt-dlm-devinfo-static: revert edge=true

KlausPopp committed:
test

KlausPopp committed:
add makeStaticDeviceInfo

KlausPopp committed:
fix module_client internal doc

KlausPopp committed:
Add fwinfo read from /etc/issue

KlausPopp committed:
kyt-dlm-devinfo-static is now an iot device

KlausPopp committed:
kyt-dlm-devinfo-static read deviceCs from config

KlausPopp committed:
Merge branch 'main' into feature/kp/dlm-devinfo-iotdev

KlausPopp committed:
kyt-dlm-devinfo-static fix topic verions -> versions

KlausPopp committed:
kyt-dlm-devinfo-static fix unit test

siredmar reviewed (changes requested):
Please clean git history (squash) and create PR from correct branch without `kp`.

KlausPopp committed:
kyt-dlm-devinfo-static fix unit test

KlausPopp committed:
Merge branch 'feature/kp/dlm-devinfo-iotdev' of github.com:ci4rail/kyt into feature/kp/dlm-devinfo-iotdev

siredmar committed:
Reviewed

log -> fmt, journald also sets time and date
removed some "debug prints"
fixed unit test -> firmwareversion -> firmwareVersion

siredmar committed:
Fixed unit test

siredmar committed:
new kyt-api-server deployment tag
